src/overlays/actors/ovl_Shot_Sun/z_shot_sun.c

```diff
@@ -119,35 +125,35 @@ void ShotSun_TriggerFairy(ShotSun* this, PlayState* play) {
     }
 }
 
-void func_80BADF0C(ShotSun* this, PlayState* play) {
+void ShotSun_UpdateFairySpawner(ShotSun* this, PlayState* play) {
     Player* player = GET_PLAYER(play);
     s32 pad;
     s32 params = this->actor.params & 0xFF;
 
     if (Math3D_Vec3fDistSq(&this->actor.world.pos, &player->actor.world.pos) > SQ(150.0f)) {
-        this->unk_1A4 = 0;
+        this->fairySpawnerState = SPAWNER_INACTIVE;
     } else {
-        if (this->unk_1A4 == 0) {
+        if (this->fairySpawnerState == SPAWNER_INACTIVE) {
             if (!(player->stateFlags2 & PLAYER_STATE2_24)) {
                 player->stateFlags2 |= PLAYER_STATE2_23;
                 return;
             } else {
-                this->unk_1A4 = 1;
+                this->fairySpawnerState = SPAWNER_WAITING;
             }
         }
-        if (this->unk_1A4 == 1) {
+        if (this->fairySpawnerState == SPAWNER_WAITING) {
             func_8010BD58(play, OCARINA_ACTION_FREE_PLAY);
-            this->unk_1A4 = 2;
```

Comment:
looking at this closer, WAITING isnt exactly right (and imo the inactive state also counts as waiting)

but, `!(player->stateFlags2 & PLAYER_STATE2_24)` is the "waiting for player to start the ocarina" condition. the else happens when player starts. it goes to state 1 for one frame, starts the ocarina, then goes to the next state

so, id say 
`SPAWNER_OCARINA_START`
`SPAWNER_OCARINA_PLAYING`

Reply:
Id also probably just say SPAWNER_OUT_OF_RANGE instead of INACTIVE, since its still actively checking the distance. I see the intent with it being "inactive from doing the other checks". but I think it makes sense to be specific about the distance check